src/main/java/org/eclipsefoundation/adopters/service/impl/DefaultAdopterService.java

```diff
@@ -0,0 +1,63 @@
+package org.eclipsefoundation.adopters.service.impl;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipsefoundation.adopters.model.AdoptedProject;
+import org.eclipsefoundation.adopters.model.Adopter;
+import org.eclipsefoundation.adopters.model.AdopterList;
+import org.eclipsefoundation.adopters.model.Project;
+import org.eclipsefoundation.adopters.service.AdopterService;
+
+import com.google.gson.Gson;
+
+import io.quarkus.cache.CacheResult;
+import io.quarkus.runtime.Startup;
+
+@Startup
+@ApplicationScoped
+public class DefaultAdopterService implements AdopterService {
+
+	private List<Adopter> adopters;
+
+	@PostConstruct
+	public void init() throws IOException {
```

Comment:
So you need to re-deploy the app on each and every change of adopters.json content? Why don't you read this file from FS and potentially reload it on change?

Reply:
So you're saying that this should be deployed as an asset through a volume, and use a file system watcher to wait for updates? Interesting, then we barely ever have to redeploy this service once its live, other than updating the volume. How would we want to maintain this? We currently see a decent amount of changes, so managing it like a secret would be pretty tedious

Reply:
a first step may be to just add it to the docker image, so that only this image needs to be rebuild / re-deployed when changes happen. It will make it easier to switch a config map later on when you'll have code ready to handle reload.

Reply:
Added! WDYT?

Reply:
(I removed caching as I'm not sure how to handle it just yet. I'll be thinking of that in the morning)

Reply:
LGTM